source/app.d

```diff
@@ -53,30 +87,58 @@ void githubHook(HTTPServerRequest req, HTTPServerResponse res)
     auto json = verifyRequest(req.headers["X-Hub-Signature"], req.bodyReader.readAllUTF8);
     if (req.headers["X-Github-Event"] == "ping")
         return res.writeBody("pong");
+    if (req.headers["X-GitHub-Event"] == "status")
+    {
+        string repoSlug = json["name"].get!string;
+        string state = json["state"].get!string;
+        // no need to trigger the checker for failure/pending
+        if (state == "success")
+        {
+            if (Clock.currTime - lastFullPRCheck >= timeBetweenFullPRChecks)
+            {
+                searchForAutoMergePrs(repoSlug);
+                lastFullPRCheck = Clock.currTime();
+            }
+        }
+        return res.writeBody("handled");
+    }
     if (req.headers["X-GitHub-Event"] != "pull_request")
         return res.writeVoidBody();
 
     auto action = json["action"].get!string;
     logDebug("#%s %s", json["number"], action);
+
     switch (action)
     {
+    case "unlabeled":
+        // for now unlabel events are ignored
+        return res.writeBody("ignored");
     case "closed":
         if (json["pull_request"]["merged"].get!bool)
             action = "merged";
         goto case;
-    case "opened", "reopened", "synchronize":
+    case "opened", "reopened", "synchronize", "labeled":
         auto repoSlug = json["pull_request"]["base"]["repo"]["full_name"].get!string;
         auto pullRequestURL = json["pull_request"]["html_url"].get!string;
         auto pullRequestNumber = json["pull_request"]["number"].get!uint;
         auto commitsURL = json["pull_request"]["commits_url"].get!string;
         auto commentsURL = json["pull_request"]["comments_url"].get!string;
-        runTask(toDelegate(&handlePR), action, repoSlug, pullRequestURL, pullRequestNumber, commitsURL, commentsURL);
+        auto isOpen = json["pull_request"]["state"].get!string == "open";
+        runTaskHelper(toDelegate(&handlePR), action, repoSlug, pullRequestURL, pullRequestNumber, commitsURL, commentsURL, isOpen);
         return res.writeBody("handled");
     default:
         return res.writeBody("ignored");
     }
 }
 
+auto runTaskHelper(Fun, Args...)(Fun fun, Args args)
+{
+    if (runAsync)
+        runTask(fun, args);
```

Comment:
Actually the async processing isn't really safe and might loose jobs, maybe we never want to do things asynchronously?